### PR TITLE
Move/simplify templates stuff

### PR DESCRIPTION
**Pull request by alanorozco**
- Move `templates` templates to `template-loader.js` with their own namespac
- Simplifies `TemplateLoader` into `fetchTemplateContentFactory` with [`_.memoize`](https://lodash.com/docs/4.17.11#memoize).

### CODE REVIEW

File: src/editor.js [hunk 1]
```diff
@@ -526,7 +468,7 @@ class Editor {
     });
 
     this.codeMirror_.on('dragover', () => this.dragover_());
-    this.codeMirror_.on('dragleave', () => this.dragleave_());
+    this.codeMirror_.on('dragleave', () => this.dragleaveOrDragEnd_());
```
Comment (alanorozco) [inline src/editor.js]: (Oh yeah, oops 😅)

File: src/editor.js [hunk 2]
```diff
@@ -576,21 +518,15 @@ class Editor {
...
     this.state_.files = files.map(name => ({
       name,
-      url: this.getTemplateFileUrl_(templateName, name),
+      url: templateFileUrl(templateName, name),
     }));
     this.state_.isFilesPanelDisplayed = true;
     this.state_.isTemplatePanelDisplayed = false;
   }
 
-  getTemplateFileUrl_(templateName, filename) {
```
Comment (alanorozco) [inline src/editor.js]: Exactly the same as imported `templateFileUrl`, using that instead.

File: src/editor.js [hunk 3]
```diff
@@ -96,15 +108,15 @@ const renderEditor = ({
...
   isFilesDragHintDisplayed = false,
   isFilesPanelDisplayed = false,
+  isFullPreview = false,
   isTemplatePanelDisplayed = false,
   previewElement,
   storyDocTemplate = '',
-  viewportId = viewportIdDefault,
   templates,
   templatesJson,
+  viewportId = viewportIdDefault,
```
Comment (alanorozco) [inline src/editor.js]: sortbyname

**Comment by shruthisanthanam** [comment]
Slightly unrelated question: will we be doing this for the other components of the editor.js file (preview, files sidebar, etc.) ?

**Comment by alanorozco** [comment]
@shruthisanthanam Yeah! I think the main thing to extract are the file upload template functions.

**Pull request merged**